OpenRA.Game/FileSystem/FileSystem.cs

```diff
@@ -243,31 +233,13 @@ public static bool Exists(string name)
 				return MountedFolders.Any(f => f.Exists(name));
 		}
 
-		static Dictionary<string, Assembly> assemblyCache = new Dictionary<string, Assembly>();
-
 		public static Assembly ResolveAssembly(object sender, ResolveEventArgs e)
 		{
 			foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
 				if (assembly.FullName == e.Name)
 					return assembly;
 
-			var frags = e.Name.Split(',');
-			var filename = frags[0] + ".dll";
-
-			Assembly a;
-			if (assemblyCache.TryGetValue(filename, out a))
-				return a;
-
-			if (Exists(filename))
-				using (var s = Open(filename))
-				{
-					var buf = s.ReadBytes((int)s.Length);
-					a = Assembly.Load(buf);
-					assemblyCache.Add(filename, a);
-					return a;
-				}
-
-			return null;
+			throw new InvalidOperationException("Unable to resolve assembly {0}!".F(e.Name));
```

Comment:
According to the docs this should continue to `return null;` instead of throwing.